core/src/network_changed/linux.rs

```diff
@@ -0,0 +1,51 @@
+use anyhow::Result;
+use futures::stream::StreamExt;
+use netlink_packet_core::NetlinkPayload::InnerMessage;
+use netlink_packet_route::{route::Nla::Gateway, RtnlMessage::NewRoute};
+use netlink_sys::{AsyncSocket, SocketAddr};
+use rtnetlink::new_connection;
+use tokio::{sync::mpsc::{self, error::TrySendError}, task::JoinHandle};
+use tracing::trace;
+
+#[must_use]
+pub struct NetworkChangedListener {
+    handle_conn: JoinHandle<()>,
+    handle_loop: JoinHandle<()>
+}
+
+impl NetworkChangedListener {
+    pub fn listen() -> Result<(Self, mpsc::Receiver<()>)> {
+        let (mut conn, _, mut messages) = new_connection()?;
+        let addr = SocketAddr::new(0, (libc::RTMGRP_IPV4_ROUTE).try_into().unwrap());
+        conn.socket_mut().socket_mut().bind(&addr)?;
+        let (tx, rx) = mpsc::channel(1);
+        let _ = tx.try_send(());
+        let handle_conn = tokio::spawn(conn);
+        let handle_loop = tokio::spawn(async move {
+            while let Some((message, _)) = messages.next().await {
```

Comment:
In Windows implementation, there is a initial message. Is it the same as Linux implementation?